### ISSUE TITLE
norm(), svd(), pinv() for complex matrices

### ISSUE THREAD

**Issue by jli05** [scalanlp/breeze#502]
Some functions are limited to Double matrices. Would it be possible to make them work with complex matrices as well? I could do a PR for pinv() but I don't know how the interface of breeze and netlib-java works, and how to delegate the call to the native library when it's present.

**Comment by dlwh** [comment]
There's a lot of awkwardness around complex numbers right now because we
box them, but BLAS of course expects them unboxed. I could just flatten the
arrays in the obvious way, but that has terrible performance... Happy to
help you implement this, but I'm generally pretty blocked on better complex
support until I have a good solution.

**Comment by jli05** [comment]
Could you keep this in mind? It'd be helpful for Randomised algorithms, sketching algorithms, some memory cell in neural networks... Thanks

**Comment by dlwh** [comment]
yeah i'm definitely interested in making this work  (or helping someone
make it work), but I was just letting you know what's blocking me (besides
general lack of time)

**Comment by jli05** [comment]
Could you outline how to extend these functions for complex matrices, how to delegate to native calls, and possibly give a simple starter code for example? Thanks in advance
